### PR TITLE
Created VA Profile Stored Function

### PR DESCRIPTION
**Pull request by maham360**
#224 #649 

Stored function returns boolean in table which correctly created and updates rows; will fix this in new ticket for function to return value

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -98,8 +98,8 @@ def update_from_original(self, original):
 
 class ServiceUser(db.Model):
     __tablename__ = 'user_to_service'
-    user_id = db.Column(UUID(as_uuid=True), db.ForeignKey('users.id'), primary_key=True)
-    service_id = db.Column(UUID(as_uuid=True), db.ForeignKey('services.id'), primary_key=True)
+    user_id = db.Column(UUID(as_uuid=True), db.ForeignKey('users.id'), primary_key=True, nullable=True)
+    service_id = db.Column(UUID(as_uuid=True), db.ForeignKey('services.id'), primary_key=True, nullable=True)
```
Comment (kalbfled) [inline app/models.py:102]: I think having a nullable primary key will cause integrity errors.

File: app/models.py [hunk 2]
```diff
@@ -399,8 +399,8 @@ class Service(db.Model, Versioned):
     go_live_user_id = db.Column(UUID(as_uuid=True), db.ForeignKey('users.id'), nullable=True)
     go_live_user = db.relationship('User', foreign_keys=[go_live_user_id])
     go_live_at = db.Column(db.DateTime, nullable=True)
-    sending_domain = db.Column(db.String(255), nullable=True, unique=False)
-    smtp_user = db.Column(db.String(255), nullable=True, unique=False)
+    sending_domain = db.Column(db.Text(255), nullable=True, unique=False)
+    smtp_user = db.Column(db.Text(255), nullable=True, unique=False)
```
Comment (kalbfled) [inline app/models.py:403]: Why this change?

File: migrations/versions/0132_add_sms_prefix_setting.py [hunk 3]
```diff
@@ -15,7 +15,7 @@
 
 def upgrade():
     op.add_column('services', sa.Column('prefix_sms', sa.Boolean(), nullable=True))
-    op.add_column('services_history', sa.Column('prefix_sms', sa.Boolean(), nullable=True))
+    op.add_column('services_history', sa.Column('prefix_sms', sa.Boolean(), nullable=True, server_default='t'))
```
Comment (kalbfled) [inline migrations/versions/0132_add_sms_prefix_setting.py:18]: Is the server default correct for Postgres?
https://www.postgresql.org/docs/11/datatype-boolean.html

**Comment by kalbfled** [comment]
I have incorporated the changes to the local cache table and stored function into branch #679.  We should incorporate additional changes in separate PRs for separate tickets.

@maham360 @jakehova

File: app/models.py (hunk 1 above)
Comment (maham360) [reply app/models.py:102]: Nullable was changed to "True" in the model to match the `user_id `and `service_id `column type we have in the database

File: app/models.py (hunk 2 above)
Comment (maham360) [reply app/models.py:403]: "String" was changed to "Text" in the model to match the `sending_domain` and `smtp_user` column type we have in the database

File: migrations/versions/0132_add_sms_prefix_setting.py (hunk 3 above)
Comment (maham360) [reply migrations/versions/0132_add_sms_prefix_setting.py:18]: I believe this is the correct syntax to set `server_default='t'`; in the docs it states "Unique prefixes of these strings are also accepted, for example t or n"
